### PR TITLE
treewide: fix typos

### PR DESCRIPTION
**Pull request by loicreynier**
Found a few typos so I ran `typos` to fix them. Most of the typos were in option descriptions. Checks passed locally.

Also added a pre-commit hook for it in the devshell. Let me know if we should keep this hook or only merge the typo fixes.

### CODE REVIEW

File: plugins/utils/neogen.nix [hunk 1]
```diff
@@ -112,53 +112,53 @@ in {
           - `"vsnip"` (https://github.com/hrsh7th/vim-vsnip)
       '';
 
-      placeholderHighligt = helpers.defaultNullOpts.mkStr "DiagnosticHint" ''
+      placeholderHighlight = helpers.defaultNullOpts.mkStr "DiagnosticHint" ''
```
Comment (traxys) [inline plugins/utils/neogen.nix:115]: We  need to introduce a warning that this option was renamed, even if it was a  typo :/ See how other renames are done

File: plugins/utils/vim-matchup.nix [hunk 2]
```diff
@@ -78,7 +78,7 @@ with lib; {
         helpers.defaultNullOpts.mkInt 60
         "Adjust timeouts in milliseconds for matchparen highlighting";
 
-      deffered = {
+      deferred = {
```
Comment (traxys) [inline plugins/utils/vim-matchup.nix:81]: You need to rename both sub options  in  this case

**Review by traxys — CHANGES_REQUESTED** [review]

**Comment by loicreynier** [comment]
Just added renames using `mkRenamedOptionModule`. It is what you intended? Haven't go the time to rebase into `main` yet.

**Comment by traxys** [comment]
Yes exactly!

**Comment by loicreynier** [comment]
Rebased into `main`.

**Review by traxys — APPROVED** [review]

**Pull request merged**